functions/package.json

```diff
@@ -3,16 +3,17 @@
   "description": "Cloud Functions for Firebase",
   "scripts": {
     "lint": "",
-    "serve": "firebase emulators:start --only functions",
-    "shell": "firebase functions:shell",
+    "serve": "npm run build && firebase emulators:start",
+    "shell": "npm run build && firebase functions:shell",
     "start": "npm run shell",
-    "deploy": "firebase deploy --only functions",
-    "logs": "firebase functions:log"
+    "deploy": "npm run build && firebase deploy",
+    "logs": "npm run build firebase functions:log",
+    "build": "tsc"
   },
   "engines": {
     "node": "18"
   },
-  "main": "index.js",
+  "main": "lib/index.js",
```

Comment:
after `tsc` compiles the code, `lib/index.js` is where the compiled index is located